src/database/schema/ChatSchema.ts

```diff
@@ -341,6 +353,67 @@ class ChatSchema implements BaseDbSchema {
     }
   };
 
+  static updateDeletedChatType = async (db: SQLiteDatabase, messageId: string) => {
+    try {
+      const selectQuery = `SELECT raw_json FROM ${ChatSchema.getTableName()} WHERE id = ?;`;
+      const resultSet = await db.executeSql(selectQuery, [messageId]);
+      const result = resultSet[0].rows.item(0);
+      const rawJson = JSON.parse(result.raw_json);
+      rawJson.message_type = MESSAGE_TYPE_DELETED;
+      rawJson.text = DELETED_MESSAGE_TEXT;
+      const updatedRawJson = JSON.stringify(rawJson);
+
+      const updateQuery = `UPDATE ${ChatSchema.getTableName()}
+        SET type = ?, message = ?, raw_json = ?
+        WHERE id = ?;`;
+
+      const updateReplacement = ['deleted', DELETED_MESSAGE_TEXT, updatedRawJson, messageId];
+
+      await db.executeSql(updateQuery, updateReplacement);
+    } catch (e) {
+      console.log('error updating deleted chat:', e);
+    }
+  };
+
+  static updateDeletedRepliedChat = async (
+    db: SQLiteDatabase,
+    channelId: string,
+    messageId: string,
+    createdAt: string
+  ) => {
+    try {
+      const selectQuery = `SELECT id, raw_json FROM ${ChatSchema.getTableName()} WHERE channel_id = ? AND created_at > ?;`;
+      const resultSet = await db.executeSql(selectQuery, [channelId, createdAt]);
+      const {rows} = resultSet[0];
+
+      const updatePromises: Promise<any>[] = [];
+
+      // eslint-disable-next-line no-plusplus
+      for (let i = 0; i < rows.length; i++) {
+        const result = rows.item(i);
+        const rawJson = JSON.parse(result.raw_json);
+
+        // Check if messageId matches
+        if (rawJson.reply_data && rawJson.reply_data.id === messageId) {
+          const newJson = {...rawJson};
+          newJson.reply_data.text = DELETED_MESSAGE_TEXT;
+          newJson.message.reply_data.text = DELETED_MESSAGE_TEXT;
+          newJson.reply_data.message_type = MESSAGE_TYPE_DELETED;
+          newJson.updated = true;
+          const updatedRawJson = JSON.stringify(newJson);
+
+          // Update the chat
+          const updateQuery = `UPDATE ${ChatSchema.getTableName()} SET raw_json = ? WHERE id = ?;`;
+          updatePromises.push(db.executeSql(updateQuery, [updatedRawJson, result.id]));
+        }
+      }
+
+      await Promise.all(updatePromises);
+    } catch (e) {
+      console.log('error updating related chat:', e);
+    }
+  };
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The addition of these two static methods, `updateDeletedChatType` and `updateDeletedRepliedChat`, is a significant improvement. They allow for updating the chat type and related chats when a chat message is deleted. This is crucial for maintaining consistency in the application's state. However, error handling could be improved by throwing the errors instead of just logging them. This would allow the calling code to decide how to handle the errors.
```diff
-      console.log('error updating deleted chat:', e);
+      throw new Error(`Error updating deleted chat: ${e}`);
...
-      console.log('error updating related chat:', e);
+      throw new Error(`Error updating related chat: ${e}`);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->